src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/Views/DynamicTableViewIconCell.swift

```diff
@@ -76,7 +77,18 @@ class DynamicTableViewIconCell: UITableViewCell {
 		iconImageView.tintColor = customTintColor ?? tintColor
 		iconImageView.image = image
 		iconImageView.isHidden = image == nil
-
+		
+		// swap label and image so the image is set to the right. Because it is initialized to the left, we need this only for .right
+		if imageAlignment == .right,
+		   let imageView = stackView.subviews.first {
+			stackView.removeArrangedSubview(imageView)
+			stackView.setNeedsLayout()
```

Comment:
Are the `setNeedsLayout` and `layoutIfNeeded` needed?
Would it be an opportunity to pass the imageAlignment at init?
If not, configure should also handle the swap back to right.

Reply:
Refactored it a bit - please have a look again @ArturFriesen.
Inject it in the init is not possible because of the inheritance of the DynamicTableViewCell. I would have to make too many changes.